src/back-end/lib/resources/proposal/team-with-us.ts

```diff
@@ -720,6 +719,20 @@ const update: crud.Update<
             });
           }
 
+          const validatedOrganization = await validateProposalOrganization(
+            connection,
+            validatedTWUProposal.value.organization?.id,
+            request.session
+          );
+          const organization = getValidValue(validatedOrganization, undefined);
+          if (!organization) {
+            return invalid({
+              proposal: adt("submit" as const, [
+                "An organization must be specified prior to submitting."
+              ])
+            });
+          }
```

Comment:
Not sure if this is the best way to check that an organization exists and the error message is possibly confusing since the user didn't submit an org.

Thoughts?

Reply:
@IanFonzie When submitting a proposal, the entire proposal should be validated, so I think this makes sense. Vendors must select a qualified org to submit a valid proposal, so the error message is helpful (IMO) as it explains this requirement.

However, it looks like `validateProposalOrganization` only checks that the organization exists in the database. It doesn't validate whether or not the organization is TWU qualified. Is this check performed elsewhere?

Reply:
Yup, lines 737 -> 757 perform additional validation on the team and provided service areas and 780 should check that they're TWU qualified.

Reply:
Got it. In that case I would use `validateOrganizationId` instead of `validateProposalOrganization` since you need the org to be defined. Is there any other use for `validateProposalOrganization`? If not, you can remove the function from the source code altogether.